pkg/netpol/eval/check_eval.go

```diff
@@ -0,0 +1,257 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package eval
+
+import (
+	"errors"
+
+	netv1 "k8s.io/api/networking/v1"
+
+	"github.com/np-guard/netpol-analyzer/pkg/internal/netpolerrors"
+	"github.com/np-guard/netpol-analyzer/pkg/netpol/eval/internal/k8s"
+)
+
+// this file contains methods for checking wether specific connection between two peers is allowed or not;
+// those funcs are related to the `eval` command
+
+// checkIfAllowedNew: (connection-set based computation) returns true if the given input connection is
+// allowed by network policies
+// currently used only for testing
+func (pe *PolicyEngine) checkIfAllowedNew(src, dst, protocol, port string) (bool, error) {
+	allowedConns, err := pe.allAllowedConnections(src, dst)
+	if err != nil {
+		return false, err
+	}
+	return allowedConns.Contains(port, protocol), nil
+}
+
+// CheckIfAllowed returns true if the given input connection is allowed by k8s: admin-network-policies,
+// network-policies and the baseline-admin-network-policy
+func (pe *PolicyEngine) CheckIfAllowed(src, dst, protocol, port string) (bool, error) {
+	srcPeer, err := pe.getPeer(src)
+	if err != nil {
+		return false, err
+	}
+	dstPeer, err := pe.getPeer(dst)
+	if err != nil {
+		return false, err
+	}
+	// cases where any connection is always allowed
+	if isPodToItself(srcPeer, dstPeer) || isPeerNodeIP(srcPeer, dstPeer) || isPeerNodeIP(dstPeer, srcPeer) {
+		return true, nil
+	}
+
+	hasResult, res := pe.cache.hasConnectionResult(srcPeer, dstPeer, protocol, port)
+	if hasResult {
+		return res, nil
+	}
+
+	egressRes, err := pe.allowedXgressConnection(srcPeer, dstPeer, false, protocol, port)
+	if err != nil {
+		return false, err
+	}
+	if !egressRes {
+		pe.cache.addConnectionResult(srcPeer, dstPeer, protocol, port, false)
+		return false, nil
+	}
+	ingressRes, err := pe.allowedXgressConnection(srcPeer, dstPeer, true, protocol, port)
+	if err != nil {
+		return false, err
+	}
+	pe.cache.addConnectionResult(srcPeer, dstPeer, protocol, port, ingressRes)
+	return ingressRes, nil
+}
+
+// allowedXgressConnection returns if the given input connection is allowed on the given ingress/egress direction
+// by k8s policies api
+func (pe *PolicyEngine) allowedXgressConnection(src, dst k8s.Peer, isIngress bool, protocol, port string) (bool, error) {
+	// first checks if the connection is allowed or denied by the admin-network-policies (return anpRes).
+	// if the connection is passed by the ANPs or not captured by them, then will continue to NPs (pass = true)
+	anpRes, pass, err := pe.allowedXgressConnectionByAdminNetpols(src, dst, isIngress, protocol, port)
```

Comment:
also here - rename var `pass` to be `passOrUncaptured`

Reply:
done